config/README.md

```diff
@@ -0,0 +1,68 @@
+# config
+
+This package is an wrapper over viper wwith opinionated defaults that allows loading config from a yaml file and environment variables.
+
+## Usage
+
+```go
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/odpf/salt/config"
+)
+
+type Config struct {
+	Port     int          `mapstructure:"port" default:"8080"`
+	NewRelic NestedConfig `mapstructure:"newrelic"`
+	LogLevel string       `mapstructure:"log_level" default:"info"`
+}
+
+type NestedConfig struct {
+	Enabled bool   `mapstructure:"enabled" default:"false"`
+	AppName string `mapstructure:"appname" default:"app"`
+	License string `mapstructure:"license"`
+}
+
+func main() {
+	var c Config
+	l := config.NewLoader(
+		// config.WithConfigName("config"),
+		// config.AddConfigPath("~/.app"),
+	)
+	l.Load(&c) // pass pointer to the struct into which you want to load config
+	s, _ := json.MarshalIndent(c, "", "  ") // spaces: 2 | tabs: 1 😛
+	fmt.Println(string(s))
+}
+```
+
+In the above program a YAML file or environment variables can be used to configure.
+
+```yaml
+port: 9000
+newrelic:
```

Comment:
Will it support a config from env that contains underscore in parent and child? For example
```
port: 9000
new_relic:
  enabled: true
  license_key: base64encodedkey
log_level: debug
```
Now to configure `license_key` what should be the ENV variable?

Reply:
The env key would be - `new_relic_license_key`.

Reply:
updated the example in the README for the same